src/rtcp/receiverReport.ts

```diff
@@ -0,0 +1,370 @@
+		}
+
+		this.buffer = buffer.slice(undefined, REPORT_LENGTH);
+	}
+
+	/**
+	 * @ignore
+	 */
+	dump(): any
+	{
+		return {
+			ssrc         : this.getSsrc(),
+			fractionLost : this.getFractionLost(),
+			totaLlost    : this.getTotalLost(),
+			lastSeq      : this.getHighestSeqNumber(),
+			jitter       : this.getJitter(),
+			lsr          : this.getLastSRTimestamp(),
+			dlsr         : this.getDelaySinceLastSR()
+		};
+	}
+
+	/**
+	 * Get the internal buffer containing the RTCP Receiver Report binary.
+	 */
+	getBuffer(): Buffer
+	{
+		return this.buffer;
+	}
+
+	/*
+	 * Get receiver SSRC.
+	 */
+	getSsrc(): number
+	{
+		return this.buffer.readUInt32BE(0);
+	}
+
+	/*
+	 * Set receiver SSRC.
+	 */
+	setSsrc(ssrc: number): void
+	{
+		this.buffer.writeUInt32BE(ssrc, 0);
+	}
+
+	/*
+	 * Get fraction lost.
+	 */
+	getFractionLost(): number
+	{
+		return this.buffer.readUInt8(4);
+	}
+
+	/*
+	 * Set fraction lost.
+	 */
+	setFractionLost(fractionLost: number): void
+	{
+		this.buffer.writeUInt8(fractionLost, 4);
+	}
+
+	/*
+	 * Get total lost.
+	 */
+	getTotalLost(): number
+	{
+		let value = this.buffer.readUIntBE(5, 3);
+
+		// Possitive value.
+		if (((value >> 23) & 1) == 0)
+		{
+			return value;
+		}
+
+		// Negative value.
+		if (value != 0x0800000)
+		{
+			value &= ~(1 << 23);
+		}
+
+		return -value;
+	}
+
+	/*
+	 * Set total lost.
+	 */
+	setTotalLost(totalLost: number): void
+	{
+		// Get the limit value for possitive and negative totalLost.
+		const clamp = (totalLost >= 0) ? totalLost > 0x07FFFFF ? 0x07FFFFF : totalLost
+			: -totalLost > 0x0800000 ? 0x0800000 : -totalLost;
+
+		const value = (totalLost >= 0) ? (clamp & 0x07FFFFF) : (clamp | 0x0800000);
+
+		this.buffer.writeUIntBE(value, 5, 3);
+	}
+
+	/*
+	 * Get highest RTP sequence number.
+	 */
+	getHighestSeqNumber(): number
+	{
+		return this.buffer.readUIntBE(8, 4);
+	}
+
+	/*
+	 * Set highest RTP sequence number.
+	 */
+	setHighestSeqNumber(lastSeq: number): void
+	{
+		this.buffer.writeUIntBE(lastSeq, 8, 4);
+	}
+
+	/*
+	 * Get interarrival jitter.
+	 */
+	getJitter(): number
+	{
+		return this.buffer.readUIntBE(12, 4);
+	}
+
+	/*
+	 * Set interarrival jitter.
+	 */
+	setJitter(jitter: number)
+	{
+		this.buffer.writeUIntBE(jitter, 12, 4);
+	}
+
+	/*
+	 * Set last Sender Report timestamp.
+	 */
+	getLastSRTimestamp(): number
+	{
+		return this.buffer.readUIntBE(16, 4);
+	}
+
+	/*
+	 * Set last Sender Report timestamp.
+	 */
+	setLastSRTimestamp(lsr: number): void
+	{
+		this.buffer.writeUIntBE(lsr, 16, 4);
+	}
+
+	/*
+	 * Get delay since last Sender Report.
+	 */
+	getDelaySinceLastSR(): number
+	{
+		return this.buffer.readUIntBE(20, 4);
+	}
+
+	/*
+	 * Set delay since last Sender Report.
+	 */
+	setDelaySinceLastSR(dlsr: number): void
+	{
+		this.buffer.writeUIntBE(dlsr, 20, 4);
+	}
+}
+
+export class ReceiverReportPacket extends RtcpPacket
+{
+	// Receiver Reports.
+	private reports: ReceiverReport[] = [];
+
+	// Packet Type.
+	static packetType = PacketType.RR;
```

Comment:
Let's put static members first:

```suggestion
	// Packet Type.
	static packetType = PacketType.RR;

	// Receiver Reports.
	private reports: ReceiverReport[] = [];
```